File: sdk/nodejs/examples/quickstart.js
const { NeuroSleepClient } = require("../dist/index");

async function main() {
  const client = new NeuroSleepClient(
    "nsn_live_your_key_here",
    "http://localhost:8001/api/v1"
  );

  console.log("🚀 Initializing NeuroSleepNet Example (Node.js)...");

  try {
    // 1. Create a task
    console.log("\n📝 Creating context 'Agent-User Learning'...");
    const task = await client.createTask("Agent-User Learning");
    const taskId = task.id;
    console.log(`✅ Task created with ID: ${taskId}`);

    // 2. Add memories
    console.log("\n🧠 Adding memories...");
    await client.addMemory(
      "The user's favorite programming language is Python and they dislike Java.",
      taskId,
      { source: "conversation_01", type: "preference" }
    );
    await client.addMemory(
      "The user is currently working on a production-ready AI platform called NeuroSleepNet.",
      taskId,
      { source: "conversation_01", type: "project" }
    );
    console.log("✅ Memories added.");

    // 3. Search with Attention scorer
    console.log("\n🔍 Searching for user preferences...");
    const results = await client.search(
      "What are the user's coding interests?",
      taskId,
      2
    );

    results.memories.forEach((res, i) => {
      console.log(`\nResult ${i + 1}:`);
      console.log(` - Content: ${res.content}`);
      console.log(` - Attention Score: ${res.attention_score.toFixed(2)}`);
      console.log(` - Explanation: ${res.why_retrieved}`);
    });

    // 4. Check dashboard stats
    console.log("\n📊 Fetching dashboard stats...");
    const stats = await client.getStats();
    console.log(` - Total Memories: ${stats.total_memories}`);
    console.log(` - Monthly Ops: ${stats.monthly_ops}`);
    console.log(` - Memory Health: ${stats.memory_health_score}%`);

    // 5. Nightly Sleep Phase (forced trigger)
    console.log("\n💤 Triggering Sleep Phase (Consolidation)...");
    const sleepResult = await client.triggerSleep();
    console.log(`✅ Sleep completed: Consolidated ${sleepResult.consolidated} memories.`);

  } catch (error) {
    console.log("\n❌ Error:", error.message);
    console.log("💡 Hint: Ensure Docker stack is up and API is responding on http://localhost:8001");
  }
}

main();

File: sdk/python/examples/quickstart.py
import os
import sys

# Add the parent directory to sys.path to allow importing from local 'neurosleepnet'
sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), "..")))

from neurosleepnet.client import NeuroSleepClient

def main():
    # 1. Initialize Client (assuming default API key or mock for now)
    client = NeuroSleepClient(
        api_key="nsn_live_your_key_here",
        base_url="http://localhost:8001/api/v1"
    )

    print("🚀 Initializing NeuroSleepNet Example...")

    try:
        # 2. Check health
        # health = client._request("GET", "/health") 
        # print(f"✅ Health Status: {health['status']}")

        # 3. Create a task
        print("\n📝 Creating context 'Agent-User Learning'...")
        task = client.create_task("Agent-User Learning")
        task_id = task["id"]
        print(f"✅ Task created with ID: {task_id}")

        # 4. Add memories
        print("\n🧠 Adding memories...")
        client.add_memory(
            content="The user's favorite programming language is Python and they dislike Java.",
            task_id=task_id,
            metadata={"source": "conversation_01", "type": "preference"}
        )
        client.add_memory(
            content="The user is currently working on a production-ready AI platform called NeuroSleepNet.",
            task_id=task_id,
            metadata={"source": "conversation_01", "type": "project"}
        )
        print("✅ Memories added.")

        # 5. Search with Attention scorer
        print("\n🔍 Searching for user preferences...")
        results = client.search(
            query="What are the user's coding interests?",
            task_id=task_id,
            top_k=2
        )

        for i, res in enumerate(results["memories"], 1):
            print(f"\nResult {i}:")
            print(f" - Content: {res['content']}")
            print(f" - Attention Score: {res['attention_score']:.2f}")
            print(f" - Explanation: {res['why_retrieved']}")

        # 6. Check dashboard stats
        print("\n📊 Fetching dashboard stats...")
        stats = client.get_stats()
        print(f" - Total Memories: {stats['total_memories']}")
        print(f" - Monthly Ops: {stats['monthly_ops']}")
        print(f" - Memory Health: {stats['memory_health_score']}%")

        # 7. Nightly Sleep Phase (forced trigger)
        print("\n💤 Triggering Sleep Phase (Consolidation)...")
        sleep_result = client.trigger_sleep()
        print(f"✅ Sleep completed: Consolidated {sleep_result['consolidated']} memories.")

    except Exception as e:
        print(f"\n❌ Error: {e}")
        print("💡 Hint: Ensure Docker stack is up and API is responding on http://localhost:8001")

if __name__ == "__main__":
    main()
